### DISCUSSION TITLE
Sequential execution of multiple APIs by providing name to each of the APIs

### DISCUSSION THREAD

**Discussion by JayasimhaNagaraj** [artilleryio/artillery#1086] [Artillery]
Hi Team,

We have a scenario which consists of 7 APIs.  All the APIs are interdependent and would require the APIs to execute sequentially.
i.e. API 1 to API 7.
However, when we are assigning the names to each of the 7 APIs, its  getting executed randomly (only one  API at a time.) 

Could you please let us know the approach in which we can name all the 7 APIs and execute all the 7 APIs sequentially in one iteration without skipping any of the APIs.

Below is the setup for reference:

 phases:
    - duration: 1
      arrivalRate: 1

============
scenarios:
  -  name: " 1.ABC"
     flow:
      - function: "generateRandomData"
      - post:
  -  name: " 2.DEF"
     flow:
      - post:

**Comment by hassy** [comment]
If API calls depend on each other and/or need to run sequentially, you can put them all into one scenario, in the order that you need them to be executed in.

**Reply by JayasimhaNagaraj** [reply]
Hi Hassy,

Thanks for your feedback.  Sure, combining of all APIs under one scenario works. 
However, was looking for an option in which we can **name all the APIs** and execute them sequentially **without skipping any of the APIs.**

**Reply by hassy** [reply]
@JayasimhaNagaraj if the API endpoints have to be hit sequentially and there are dependencies between them, then they should be in the same scenario. Having more than one scenario lets you emulate different users doing different (and independent) things on the system.

Not sure what you mean by naming the API - could you elaborate?

**Reply by JayasimhaNagaraj** [reply]
Hi @hassy 👍 

Thanks for your inputs. Got more clarity regarding my query :)
